src/blorb.c

```diff
@@ -214,20 +223,31 @@ uint8_t *loadpng(char *fname, uint32_t *p_size, int *p_width, int *p_height) {
 		exit(1);
 	}
 	if(st.st_size > 0x1000000UL) {
-		report(LVL_ERR, 0, "PNG file too large: %s", fname);
+		report(LVL_ERR, 0, "Image file %s is too large: %lu bytes (maximum is %lu)", fname, st.st_size, 0x1000000UL);
 		exit(1);
 	}
+	
 	size = (uint32_t) st.st_size;
 	buf = malloc(size);
 	fread(buf, size, 1, f);
 	fclose(f);
-
-	if(size < 8 || memcmp(buf, magic, 8)) {
-		report(LVL_ERR, 0, "Not a PNG file: %s", fname);
+	
+	if(size < 10) {
+		report(LVL_ERR, 0, "Image file %s is too small and likely corrupt: %d bytes", fname, size);
 		exit(1);
 	}
-
-	*p_size = size;
+	
+	*fsize = size;
+	return buf;
+}
+// The following files then handle different formats it could be in
```

Comment:
```suggestion
// The following functions then handle different formats it could be in
```

Unless I'm misunderstanding what you're getting at here!